ui/src/app/components/add-new-column/add-new-column.component.html

```diff
@@ -21,6 +21,28 @@ <h2>Column Details</h2>
             </mat-form-field>
             <br>
         </div>
+        <div *ngIf="selectedDatatype != '' && autoGenMap[selectedDatatype].length > 1 && srcDbName == 'MySQL'">
+            <mat-form-field class="full-width" appearance="outline">
+                <mat-label>Auto Generated</mat-label>
+                <mat-select [(ngModel)]="selectedAutoGen" class="input-field" formControlName="autoGen">
+                    <ng-container *ngFor="let autoGen of autoGenMap[selectedDatatype]">
+                        <mat-option *ngIf="autoGen.GenerationType === ''" [value]="autoGen">
+                            {{ autoGen.Name !== '' ? autoGen.Name : 'None' }}
+                        </mat-option>
+                    </ng-container>
+                    <ng-container *ngIf="processedAutoGenMap[selectedDatatype] as types">
+                        <ng-container *ngFor="let type of types | keyvalue">
+                            <mat-optgroup *ngIf="type.key !== ''" [label]="type.key">
+                                <mat-option *ngFor="let autoGen of type.value" [value]="autoGen">
+                                    {{ autoGen.Name }}
```

Comment:
Does this list refresh/update as the user creates new sequences?

Currently this only supports for UUID so this will not arise, but please check if we need to use any observables etc. here - the end goal is that as user is creating new custom sequences, this list should be automatically updated.

Reply:
So this is list is created from getAutoGenMap. I will refresh it by updating the auogenMap value